addons/misra.py

```diff
@@ -340,30 +340,37 @@ def isStdLibId(id_, standard='c99'):
 
 # Reserved keywords defined in ISO/IEC9899:1990 -- ch 6.1.1
 C90_KEYWORDS = {
-    'auto', 'break', 'double', 'else', 'enum', 'extern', 'float', 'for',
-    'goto', 'if', 'case', 'char', 'const', 'continue', 'default', 'do', 'int',
-    'long', 'struct', 'switch', 'register', 'typedef', 'union', 'unsigned',
-    'void', 'volatile', 'while', 'return', 'short', 'signed', 'sizeof',
-    'static'
+    'auto', 'break', 'case', 'char', 'const', 'continue', 'default', 'do',
+    'double', 'else', 'enum', 'extern', 'float', 'for', 'goto', 'if', 
+    'int', 'long', 'register', 'return', 'short', 'signed',
+    'sizeof', 'static', 'struct', 'switch', 'typedef', 'union', 'unsigned',
+    'void', 'volatile', 'while'
 }
 
 
 # Reserved keywords defined in ISO/IEC 9899 WF14/N1256 -- ch. 6.4.1
-C99_KEYWORDS = {
-    'auto', 'break', 'case', 'char', 'const', 'continue', 'default', 'do',
-    'double', 'else', 'enum', 'extern', 'float', 'for', 'goto', 'if', 'inline',
-    'int', 'long', 'register', 'restrict', 'return', 'short', 'signed',
-    'sizeof', 'static', 'struct', 'switch', 'typedef', 'union', 'unsigned',
-    'void', 'volatile', 'while', '_Bool', '_Complex', '_Imaginary'
+C99_ADDED_KEYWORDS = {
+    'inline', 'restrict', '_Bool', '_Complex', '_Imaginary',
+    'bool', 'complex', 'imaginary',
 }
 
+C11_ADDED_KEYWORDS = {
```

Comment:
Out of interest, why do you need this?

The point is that MISRA 2012 supports C90 and C99 only. This means, that other constructions added in C11 can not be handled by MISRA rules, so the codebase that uses them can not be MISRA compatible.

Reply:
When you look at line 333, isStdLibId, the c11 standard is covered.
But if you pass c11 argument to this function originally, it would use an empty list. 

Also according cppcheck.sourceforge.io, C++17 support was added in cppcheck (2021-04-26) so why not adding at least the keywords to the misra knowledge?

My project is using c11 standard, and the isKeyord function is working correct, with c11. So I wanted to add c11 support, and make sure it would use c11 list of keywords when you would for instance try to use c17 standard. When you don't give an argument, it would still use c99 standard was before. So I don't see why it would create problems.

Reply:
Well, this was added in https://github.com/danmar/cppcheck/commit/4baf221a6b47d6944b9ea7d5c255076c060f9789. I think, @danmar have some reason for that, may be this is not just for debugging purposes.

These changes looks good for me, if they won't change default behaviour of the addon, e.g. fallback to `c99` if the user didn't set the greater standard explicitly.

Because, again, C11 and C17 code bases can not be MISRA compatible.

Reply:
I think it is good that this utility function handles c11.. probably it should be moved to cppcheckdata.py someday.